plugins/modules/mysql_role.py

```diff
@@ -0,0 +1,784 @@
+        self.cursor.execute(*query)
+        return self.cursor.fetchone()[0] > 0
+
+    def add(self, users, privs, check_mode=False, admin=False,
+            set_default_role_all=True):
+        """Add a role.
+
+        Args:
+            users (list): Role members.
+            privs (str): String containing privileges.
+            check_mode (bool): If True, just checks and does nothing.
+            admin (list): Role's admin. Contains (username, hostname).
+            set_default_role_all (bool): If True, runs SET DEFAULT ROLE ALL TO each member.
+
+        Returns:
+            bool: True if the state has changed, False if has not.
+        """
+        if check_mode:
+            if not self.exists:
+                return True
+            return False
+
+        if not admin:
+            self.cursor.execute('CREATE ROLE %s', (self.name,))
+        else:
+            query = 'CREATE ROLE %s WITH ADMIN %s@%s'
+            admin_user = admin[0]
+            admin_host = admin[1]
+            self.cursor.execute(query, (self.name, admin_user, admin_host))
+
+        if users:
+            self.add_members(users, set_default_role_all=set_default_role_all)
+
+        if privs:
+            for db_table, priv in iteritems(privs):
+                privileges_grant(self.cursor, self.name, self.host,
+                                 db_table, priv, tls_requires=None,
+                                 maria_role=self.is_mariadb)
+
+        return True
+
+    def drop(self, check_mode=False):
+        """Drop a role.
+
+        Args:
+            check_mode (bool): If True, just checks and does nothing.
+
+        Returns:
+            bool: True if the state has changed, False if has not.
+        """
+        if not self.exists:
+            return False
+
+        if check_mode and self.exists:
+            return True
+
+        self.cursor.execute('DROP ROLE %s', (self.name,))
+        return True
+
+    def add_members(self, users, check_mode=False, append_members=False,
+                    set_default_role_all=True):
+        """Add users to a role.
+
+        Args:
+            users (list): Role members.
+            check_mode (bool): If True, just checks and does nothing.
+            append_members (bool): If True, adds new members passed through users
+                not touching current members.
+            set_default_role_all (bool): If True, runs SET DEFAULT ROLE ALL TO each member.
+
+        Returns:
+            bool: True if the state has changed, False if has not.
+        """
+        if not users:
+            return False
+
+        changed = False
+        for user in users:
+            if user not in self.members:
+                if check_mode:
+                    return True
+
+                if not self.is_mariadb:
+                    self.cursor.execute('GRANT %s@%s TO %s@%s', (self.name, self.host, user[0], user[1]))
```

Comment:
I guess you can avoid this `if` (and maybe others) by using `self.full_name` to refer to the role name

Reply:
Yes, we can but it's better to pass variables through parameters whenever possible.